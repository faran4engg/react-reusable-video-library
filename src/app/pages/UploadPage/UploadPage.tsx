import { FC } from 'react';
import { Helmet } from 'react-helmet-async';
import UploadVideoFormContainer from 'app/domains/Upload/containers/UploadVideoForm';
import { UserUploadForm } from 'app/domains/Upload/components';

const UploadPage: FC = () => {
  return (
    <>
      <Helmet>
        <title>Upload a Video ðŸŽ¥</title>
        <meta name="description" content="upload video page" />
      </Helmet>
      <div className="flex justify-center items-center">
        <UploadVideoFormContainer>
          {props => <UserUploadForm {...props} />}
        </UploadVideoFormContainer>
      </div>
    </>
  );
};

export default UploadPage;
