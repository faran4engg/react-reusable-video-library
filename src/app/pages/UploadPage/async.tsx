import { lazyLoad } from 'utils/loadable';

export const UploadPage = lazyLoad(
  () => import('./UploadPage'),
  module => module.default,
);
