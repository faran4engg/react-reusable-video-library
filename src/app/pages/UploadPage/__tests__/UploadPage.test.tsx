import React from 'react';

import { create } from 'react-test-renderer';
import UploadPage from '..';

describe('<UploadPage />', () => {
  const renderComponent = () => create(<UploadPage />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Google Login button', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });
});
