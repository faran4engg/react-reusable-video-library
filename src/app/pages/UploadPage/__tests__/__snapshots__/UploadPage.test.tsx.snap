// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadPage /> should render Google Login button 1`] = `
Array [
  "<Helmet />",
  <div
    className="flex justify-center items-center"
  >
    <div
      className="w-full md:w-3/4 lg:w2/4 bg-light-mode dark:bg-dark-mode text-pink-600 font-semibold rounded-lg shadow-lg p-4"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique, labore suscipit minus iusto voluptas maxime ea sed dolor rem eaque quibusdam harum distinctio tenetur quasi amet dolores laborum blanditiis possimus?
      <div
        className="mt-4"
      >
        <button
          className="inline-flex justify-center items-center p-2 w-full font-medium leading-5 align-bottom rounded-lg shadow-md cursor-pointer  focus:outline-none dark:bg-dark-mode-content-bg hover:bg-gray-100 dark:hover:bg-gray-900"
          onClick={[Function]}
          type="submit"
        >
          <span
            className="text-gray-700 dark:text-dark-mode-text"
          >
            Login with
          </span>
          <span>
             
          </span>
          <span>
            <svg>
              google.svg
            </svg>
          </span>
          <span>
             
          </span>
          <span
            className="text-gray-700 dark:text-dark-mode-text"
          >
            to Upload
          </span>
        </button>
      </div>
    </div>
  </div>,
]
`;
